api/python/quilt3/data_transfer.py

```diff
@@ -245,19 +256,16 @@ def _upload_file(ctx, size, src_path, dest_bucket, dest_key):
         )
         upload_id = resp['UploadId']
 
-        adjuster = ChunksizeAdjuster()
-        chunksize = adjuster.adjust_chunksize(s3_transfer_config.multipart_chunksize, size)
-
-        chunk_offsets = list(range(0, size, chunksize))
+        chunk_ranges = get_chunk_ranges(size)
 
         lock = Lock()
-        remaining = len(chunk_offsets)
+        remaining = len(chunk_ranges)
         parts = [None] * remaining
 
         def upload_part(i, start, end):
             nonlocal remaining
             part_id = i + 1
-            with OSUtils().open_file_chunk_reader(src_path, start, end-start, [ctx.progress]) as fd:
+            with OSUtils().open_file_chunk_reader(src_path, start, end - start + 1, [ctx.progress]) as fd:
```

Comment:
What is the purpose of adding the +1 here? Was that a bug in the old code? Or, is there a reason that that this change requires an interval end 1 byte farther?

Reply:
See my comment above; I think the `end` value should be adjusted.